arb/dump_file.c

```diff
@@ -0,0 +1,31 @@
+/*
+    Copyright (C) 2019 Julian Rüth
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdio.h>
+#include <string.h>
+
+#include "arb.h"
+
+int arb_dump_file(FILE* stream, const arb_t x)
+{
+    int nwrite;
+    char* pos;
+    char* data = arb_dump_str(x);
+    /* replace all ' ' with '$' */
+    for (pos = data; (pos = strchr(pos, ' ')) != NULL; *pos = '$');
```

Comment:
I implemented the FILE versions by falling back to the char* ones. To be able to read until whitespace I have to escape existing whitespace. I'm not terribly proud of this but otherwise the loader functions would all need to know how many whitespaces to expect inside a block.

Reply:
(without this escaping, I also cannot share code; that's what I would like to do see https://github.com/fredrik-johansson/arb/pull/286/files#r331434364)